### PR TITLE
Refactored courses and course_categories for Django Rest Framework

### PR DESCRIPTION
**Pull request by timothy-huang**

### CODE REVIEW

File: course/views.py [hunk 1]
```diff
@@ -1,26 +1,33 @@
...
+    serializer_class = CourseCategorySerializer
 
-
-def get_courses_for_category(request, category_id):
-    course_list = get_list_or_404(Course, course_category=category_id)
-    return JsonResponse({
-        "data": course_list
-    })
+    """
+    Extra route that gets courses by category id
```
Comment (timothy-huang) [inline course/views.py:25]: This method of adding a custom route may not be officially supported anymore - saw it in a documentation site for "Django Rest Framework 2", will look into other ways to achieve the same thing based on regular site but seems to work fine for now

Comment (timothy-huang) [reply course/views.py:25]: alternative seems to be using "action" decorator, will test out on my local repo and see how it works

File: account/models.py [hunk 2]
```diff
@@ -5,80 +5,80 @@
```
Comment (timothy-huang) [inline account/models.py:5]: My autoformatter does 4 spaces for tab. Is there a clean way for us to ignore/resolve these formatting diffs?

Comment (jerrylinew) [reply account/models.py:5]: I'll change mine to 4 spaces

**Review by jerrylinew — APPROVED** [review]

**Pull request merged**